pkgs/build-support/node/fetch-npm-deps/src/parse/mod.rs

```diff
@@ -42,16 +43,19 @@ pub fn lockfile(
         .iter()
         .filter(|p| matches!(p.specifics, Specifics::Git { .. }))
     {
-        let dir = match &pkg.specifics {
-            Specifics::Git { workdir } => workdir,
-            Specifics::Registry { .. } => unimplemented!(),
+        let Specifics::Git {
+            workdir,
+            lockfile_path,
+        } = &pkg.specifics
+        else {
+            unimplemented!()
```

Comment:
Nit: I know it's what was there before this patch, but semantically this looks like `unreachable!` rather than `unimplemented!`.

Reply:
I've changed it to do the filtering and destructuring at the same time.